lib/request-handler.js

```diff
@@ -35,6 +36,31 @@ module.exports = function processRequest (options, request, response) {
 
     let shouldWriteHead = true;
     let index = 0;
+    let assetsToPreload = '';
+
+    // Early preloading in browsers to improve the primary fragments
+    const getAssetsToPreload = (fragmentReq, fragment, statusCode, headers) => {
+        // Do not preload assets for non primary fragments
+        if (!fragment.primary) {
+            return;
+        }
+        const links = parseLinkHeader(headers.link);
+        // Handle Server rendered fragments without depending on assets
+        if (links == null) {
+            return;
+        }
+        if (links.stylesheet && links.stylesheet.url) {
+            assetsToPreload += `<${links.stylesheet.url}>; rel="preload"; as="style",`;
+        }
+        if (links['fragment-script'] && links['fragment-script'].url) {
+            assetsToPreload += `<${links['fragment-script'].url}>; rel="preload"; as="script",`;
+        }
+        // Remove the event listener once we are done capturing the necessary assets
+        // to make sure we are not exceeding EventListener's limit (10 default)
+        this.removeListener('fragment:response', getAssetsToPreload);
+    };
+
+    this.on('fragment:response', getAssetsToPreload);
```

Comment:
although it gets fired for non primary stuff too.. so this actually wouldn't work.